content/guides/caching.md

```diff
@@ -4,284 +4,358 @@ sort: 11
 
-## Generating unique hashes for each file
+?> Add bash output
 
-What if we could produce the same filename, if the contents of the file did not change between builds? For example, it would be unnecessary to re-download a vendor file, when no dependencies have been updated, only application code.
+This is because webpack includes certain boilerplate, specifically the runtime and manifest, in the entry chunk.
 
-webpack allows you to generate hashes depending on file contents, by replacing the placeholder `[hash]` with `[chunkhash]`. Here is the updated config:
 
-```diff
-module.exports = {
-  /*...*/
-  output: {
-    /*...*/
--   filename: "[name].[hash].js"
-+   filename: "[name].[chunkhash].js"
-  }
-};
-```
+## Extracting Boilerplate
 
-This config will also create 2 files, but in this case, each file will get its own unique hash.
-
-```bash
-Hash: cfba4af36e2b11ef15db
-Version: webpack 2.2.0
-Time: 66ms
-                         Asset     Size  Chunks             Chunk Names
-vendor.50cfb8f89ce2262e5325.js  2.58 kB       0  [emitted]  vendor
-  main.70b594fe8b07bcedaa98.js  2.57 kB       1  [emitted]  main
-   [0] ./src/index.js 63 bytes {1} [built]
-   [1] ./src/vendor.js 63 bytes {0} [built]
-```
+As we learned in [code splitting](/guides/code-splitting), the [`CommonsChunkPlugin`](/plugins/commons-chunk-plugin) can be used to split modules out into separate bundles. A lesser-known feature of the `CommonsChunkPlugin` is extracting webpack's boilerplate and manifest which can change with every build. By specifying a name not mentioned in the `entry` configuration, the plugin will automatically extract what we want into a separate bundle:
 
-T> Don’t use [chunkhash] in development since this will increase compilation time. Separate development and production configs and use [name].js for development and [name].[chunkhash].js in production.
+__webpack.config.js__
 
+``` diff
+  const path = require('path');
++ const webpack = require('webpack');
+  const CleanWebpackPlugin = require('clean-webpack-plugin');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
 
-## Get filenames from webpack compilation stats
-
-When working in development mode, you just reference JavaScript files by entry point name in your HTML.
-
-```html
-<script src="vendor.js"></script>
-<script src="main.js"></script>
+  module.exports = {
+    entry: './src/index.js',
+    plugins: [
+      new CleanWebpackPlugin(['dist']),
+      new HtmlWebpackPlugin({
+        title: 'Caching'
+-     })
++     }),
++     new webpack.optimize.CommonsChunkPlugin({
++       name: 'runtime'
++     })
+    ],
+    output: {
+      filename: '[name].[contenthash].js',
+      path: path.resolve(__dirname, 'dist')
+    }
+  };
 ```
 
-Although, each time we build for production, we’ll get different file names. Something, that looks like this:
+Now let's run a build to see if it worked:
```

Comment:
By reading this sentence it's not clear if it worked or not and you're not clarifying it after the code block. I'd rephrase by making it less ambiguous.